### PR TITLE
Update deploy_cfns.rb

### PR DESCRIPTION
**Pull request by aneeshkp**
My changes to amazon deployment, creating 3 interfaces and reading nics and properties from nsd

### CODE REVIEW

File: scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb [hunk 1]
```diff
@@ -89,27 +90,65 @@ def base_template(name)
...
-  template_content["Resources"].merge!(template_interface(nic1, 'subnet-ac904787', security_group_1))
-  template_content["Resources"].merge!(template_interface(nic2, 'subnet-1852bb33', security_group_2)) 
-  template_content["Resources"].merge!(instance("Ec2Instance", [nic1, nic2], "EmsRefreshSpec-KeyPair", 'ami-5769193e'))
+  nic = Array.new 
+  
+  network=$evm.vmdb('ManageIQ_Providers_Amazon_NetworkManager_CloudNetwork').find_by_name(vpc_id.strip)
+  image=$evm.vmdb('ManageIQ_Providers_Amazon_CloudManager_Template').find_by_name(image_id.strip)
+  if network==nil || image==nil
+    $evm.log(:error,"vpc_id #{vpc_id} or image id #{image_id} for amazon is not valid.")
+     exit MIQ_ERROR
```
Comment (Ladas) [inline scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb:108]: we would also need to set $evm.root['ae_result'] = 'error' here

Best is to just call 

raise "vpc_id #{vpc_id} or image id #{image_id} for amazon is not valid."

then the logic for exiting state machine is at the bottom

**Comment by Ladas** [comment]
@aneeshkp awesome, I'll put it in, so I can also test the amazon. Once you will be at it, try to do the correct error handling as I commented

File: scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb [hunk 2]
```diff
@@ -89,27 +90,65 @@ def base_template(name)
...
+  availability_zone=nsd_properties['availabilityzone']
+  key_name=nsd_properties['key_name']
+  instance_type=nsd_properties['image_type']
   template_content = base_template(name)
-  template_content["Resources"].merge!(security_group(security_group_1, vpc_id))
-  template_content["Resources"].merge!(security_group(security_group_2, vpc_id))
-  template_content["Resources"].merge!(template_interface(nic1, 'subnet-ac904787', security_group_1))
-  template_content["Resources"].merge!(template_interface(nic2, 'subnet-1852bb33', security_group_2)) 
-  template_content["Resources"].merge!(instance("Ec2Instance", [nic1, nic2], "EmsRefreshSpec-KeyPair", 'ami-5769193e'))
+  nic = Array.new
```
Comment (Ladas) [inline scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb:102]: in ruby you can do

nic = []

File: scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb [hunk 3]
```diff
@@ -89,27 +90,65 @@ def base_template(name)
...
+      $evm.log(:info ,"finding #{name}  and #{value}")
+      if value!=nil
+        #check if manageIQ database has the interface and get it's id
+        $evm.log(:info ,"finding #{value} subnet in amazon")
+        subnet=$evm.vmdb('ManageIQ_Providers_Amazon_NetworkManager_CloudSubnet').find_by_name(value.strip)
+        if subnet==nil
+          $evm.log(:error,"#{value} subnet not found in amazon")
+          exit MIQ_ERROR
+        end
+          nic<< value.gsub("\s", "").gsub("_", "")
```
Comment (Ladas) [inline scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb:133]: indent to left, for the below code too

File: scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb (hunk 1 above)
Comment (aneeshkp) [inline scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb:108]: Thanks , I make those changes.

**Pull request merged**